From 542539c584656144c05ac63c3c7cc7071ab37d78 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:49:48 -0400
Subject: [tools] lib/traceevent: Several cleanups for function plugin

Message-id: <1408953210-26343-101-git-send-email-jolsa@redhat.com>
Patchwork-id: 88815
O-Subject: [PATCH RHEL7.1 BZ1133083 100/322] tools lib traceevent: Several cleanups for function plugin
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 39956e78010645ee9d121e6a6eb6d9892e3fc92e
Author: Jiri Olsa <jolsa@redhat.com>
Date: Tue Dec 3 14:09:39 2013 +0100

description
===========
Several cleanups suggested by Namhyung:

  * Remove index field from struct func_stack as it's not needed.

  * Rename get_index into add_and_get_index.

  * Use '%*X' format string capability instead of the loop
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/plugin_function.c b/tools/lib/traceevent/plugin_function.c
index 8deb22e..87acf9c 100644
--- a/tools/lib/traceevent/plugin_function.c
+++ b/tools/lib/traceevent/plugin_function.c
@@ -25,7 +25,6 @@
 #include "event-utils.h"
 
 static struct func_stack {
- int index;
  int size;
  char **stack;
 } *fstack;
@@ -57,7 +56,7 @@ static void add_child(struct func_stack *stack, const char *child, int pos)
  stack->stack[pos] = strdup(child);
 }
 
-static int get_index(const char *parent, const char *child, int cpu)
+static int add_and_get_index(const char *parent, const char *child, int cpu)
 {
  int i;
 
@@ -97,7 +96,7 @@ static int function_handler(struct trace_seq *s, struct pevent_record *record,
  unsigned long long pfunction;
  const char *func;
  const char *parent;
- int i, index;
+ int index;
 
  if (pevent_get_field_val(s, event, "ip", record, &function, 1))
   return trace_seq_putc(s, '!');
@@ -109,10 +108,9 @@ static int function_handler(struct trace_seq *s, struct pevent_record *record,
 
  parent = pevent_find_function(pevent, pfunction);
 
- index = get_index(parent, func, record->cpu);
+ index = add_and_get_index(parent, func, record->cpu);
 
- for (i = 0; i < index; i++)
-  trace_seq_printf(s, "   ");
+ trace_seq_printf(s, "%*s", index*3, "");
 
  if (func)
   trace_seq_printf(s, "%s", func);
-- 
1.7.1